File: test-apps/test-play-java-2.8/app/controllers/StreamApplication.java
package controllers;

import java.io.File;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Optional;
import java.util.Random;
import java.util.concurrent.TimeUnit;

import akka.japi.Pair;
import akka.stream.OverflowStrategy;
import akka.stream.javadsl.Source;
import akka.util.ByteString;
import play.http.HttpEntity.Streamed;
import play.mvc.Controller;
import play.mvc.Result;

public class StreamApplication extends Controller {
    private File tempFile = generateFile();
    private Source<ByteString, ?> source = Source.unfold(0, i -> { 
            if (i < 100) { 
                Thread.sleep(100); 
                return Optional.of(new Pair<Integer, ByteString>(i + 1, ByteString.fromString(i + " XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"))); 
            } else { 
                return Optional.empty(); 
            } 
        });
    
            
            
    
//    public Result index() {
//        
//    }
    public Result streamFile() {
        return ok(tempFile);
    }
    
    public Result streamChunked() {
        return ok().chunked(source);
    }
    
    public Result streamEntity() {
        return ok().sendEntity(new Streamed(source, Optional.empty(), Optional.empty()));
    }
    
    private File generateFile() {
        try {
            File file = File.createTempFile("test-post-multipart.txt", null);
            try (
                 PrintWriter printWriter = new PrintWriter(file);
            ) {
            
                Random random = new Random();
                for (int i = 0 ; i < 10000; i ++) {
                    printWriter.write(random.nextInt(10));
                }
                
                printWriter.flush();
                return file;
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        
        return null;
    }
}

File: test-apps/test-play-scala-2.6/app/controllers/StreamApplication.scala
package controllers

import java.io.File
import java.io.PrintWriter
import scala.util.Random
import scala.concurrent.ExecutionContext.Implicits.global
import akka.stream.scaladsl.Source
import akka.util.ByteString
import play.api.http.HttpEntity.Streamed
import play.api.mvc.Action
import play.api.http.FileMimeTypes
import play.api.mvc.BaseController
import javax.inject.Inject
import play.api.mvc.ControllerComponents
import play.api.mvc.AbstractController


class StreamApplication @Inject()(cc: ControllerComponents) extends AbstractController(cc) {
  val tempFile = generateFile()
  val stringChunk = ByteString.fromString("lolololololol")
  val chunkSource = Source.unfold(0) { i => if (i < 100) { Thread.sleep(100); Some(i + 1, stringChunk) } else { None } }
  
  def streamFile = Action {
    Ok.sendFile(tempFile)
  }
  
  def streamEntity = Action {
    Ok.sendEntity(Streamed(chunkSource, None, None))
  }
  
  def streamChunked = Action {
    Ok.chunked(chunkSource)
  }
  
  
  private def generateFile(): File = {
    val file = File.createTempFile("test-post-multipart", null)
    val printWriter = new PrintWriter(file)
    (1 to 10000).foreach { _ =>
      printWriter.write(Random.nextString(10))
    }

    printWriter.flush()
    
    file
  }
  
}